server/executor/assertion_converter.go

```diff
@@ -0,0 +1,57 @@
+package executor
+
+import (
+	"fmt"
+
+	"github.com/kubeshop/tracetest/assertions"
+	"github.com/kubeshop/tracetest/assertions/comparator"
+	"github.com/kubeshop/tracetest/openapi"
+)
+
+func ConvertAssertionsIntoTestDefinition(openapiAssertions []openapi.Assertion) (assertions.TestDefinition, error) {
+	testDefinition := make(assertions.TestDefinition, 0)
+	selector := ""
+	for _, assertion := range openapiAssertions {
+		selector = assertion.Selector
+		newAssertions := make([]assertions.Assertion, 0, len(openapiAssertions))
+		for _, spanAssertion := range assertion.SpanAssertions {
+			comparator, err := getComparator(spanAssertion.Operator)
+			if err != nil {
+				return assertions.TestDefinition{}, err
+			}
+			newAssertion := assertions.Assertion{
+				ID:         assertion.AssertionId,
+				Attribute:  spanAssertion.PropertyName,
+				Comparator: comparator,
+				Value:      spanAssertion.ComparisonValue,
+			}
+
+			newAssertions = append(newAssertions, newAssertion)
+		}
+		testDefinition[assertions.SpanQuery(selector)] = newAssertions
+	}
+	return testDefinition, nil
+}
+
+func getComparator(operatorName string) (comparator.Comparator, error) {
+	operationName := "="
```

Comment:
The idea was to remove the switch. If you need to use the words instead of the simbols, feel free to change the comparators to use that instead.

Reply:
As discussed offline, it makes sense to keep this switch now to keep compatibility with both the current FE implementation and the newer selectors format. We'll remove this later when we better integrate the FE with the new selectors